### PR TITLE
61804064 add fluidigm support to warehouse

### PR DESCRIPTION
**Pull request by ke4**

### CODE REVIEW

File: config/routing_keys.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ laboratory_routing_keys: &LABORATORY
   - '*.*.plate.*'
   - '*.*.labellable.*'
   - '*.*.filterpaper.*'
+  - '*.*.fluidigm.*'
   # Bulk resource
```
Comment (llh1) [inline config/routing_keys.yml:13]: As you added snp asset, you should support the snp asset routing key here, no?

File: db/migrations/012_add_fluidigm_and_snp_asset_table.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    end
+
+    # helper
+    create_table :snp_asset_container_helper do
+      primary_key :internal_id
+      String :snp_asset_uuid, :fixed => true, :size => 64
+      String :container_uuid, :fixed => true, :size => 64
+      index :container_uuid
+    end
+  end
```
Comment (llh1) [inline db/migrations/012_add_fluidigm_and_snp_asset_table.rb:53]: Where is this table used?

File: config/routing_keys.yml [hunk 3]
```diff
@@ -18,6 +20,8 @@ laboratory_routing_keys: &LABORATORY
   - '*.*.swapsamples.*'
   - '*.*.transfermultiplefilterpaperstotubes.*'
   - '*.*.transfermultiplefilterpaperstowells.*'
+  - '*.*.transferplatestofluidigm.*'
+  - '*.*.movesnpassays.*'
```
Comment (llh1) [inline config/routing_keys.yml:24]: Does this message work directly out of the box with the default code of the warehouse builder?

File: spec/lims-warehousebuilder/decoders/fluidigm_decoder_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+require 'lims-warehousebuilder/decoders/spec_helper'
+require 'lims-warehousebuilder/json_decoder'
+require 'lims-warehousebuilder/model'
+
+module Lims::WarehouseBuilder
+  describe "Decoder::FluidigmDecoder" do
```
Comment (llh1) [inline spec/lims-warehousebuilder/decoders/fluidigm_decoder_spec.rb:6]: Spec for SnpAssays ?

File: spec/lims-warehousebuilder/models/fluidigm_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
+require 'lims-warehousebuilder/models/spec_helper'
+
+module Lims::WarehouseBuilder
+  describe "Model::Fluidigm" do
```
Comment (llh1) [inline spec/lims-warehousebuilder/models/fluidigm_spec.rb:4]: Spec for SnpAssays?

File: config/routing_keys.yml [hunk 6]
```diff
@@ -18,6 +20,8 @@ laboratory_routing_keys: &LABORATORY
   - '*.*.swapsamples.*'
   - '*.*.transfermultiplefilterpaperstotubes.*'
   - '*.*.transfermultiplefilterpaperstowells.*'
+  - '*.*.transferplatestofluidigm.*'
```
Comment (llh1) [inline config/routing_keys.yml:23]: Same question as for movesnpassays below: does it work? Did you test that those messages works fine in the warehouse builder? Might be interesting to find a way to be able to test it in the specs somehow.

**Comment by ke4** [comment]
S2 app is going to be retire, so this change is not important anymore

**Pull request closed without merging**